go/private/rules/nogo.bzl

```diff
@@ -140,11 +140,41 @@ def nogo(name, visibility = None, **kwargs):
 def nogo_wrapper(**kwargs):
     if kwargs.get("vet"):
         kwargs["deps"] = kwargs.get("deps", []) + [
-            Label("@org_golang_x_tools//go/analysis/passes/atomic:go_default_library"),
-            Label("@org_golang_x_tools//go/analysis/passes/bools:go_default_library"),
-            Label("@org_golang_x_tools//go/analysis/passes/buildtag:go_default_library"),
-            Label("@org_golang_x_tools//go/analysis/passes/nilfunc:go_default_library"),
-            Label("@org_golang_x_tools//go/analysis/passes/printf:go_default_library"),
+            Label("@org_golang_x_tools//go/analysis/passes/{}".format(p))
+            for p in [
+                "appends",
+                "asmdecl",
+                "assign",
+                "atomic",
+                "bools",
+                "buildtag",
+                "cgocall",
+                "composites",
+                "copylocks",
+                "defers",
+                "directive",
+                "errorsas",
+                "framepointer",
+                "httpresponse",
+                "ifaceassert",
+                "loopclosure",
+                "lostcancel",
+                "nilfunc",
+                "printf",
+                "shift",
+                "sigchanyzer",
+                "slog",
+                "stdmethods",
+                "stringintconv",
+                "structtag",
+                "testinggoroutine",
+                "tests",
+                "timeformat",
+                "unmarshal",
+                "unreachable",
+                "unsafeptr",
+                "unusedresult",
```

Comment:
This list differs from the other one. Is this intentional? Could we share a common Starlark constant?

Reply:
Correct.

TOOLS_NOGO document said:

> # TOOLS_NOGO is a list of all analysis passes in
> # golang.org/x/tools/go/analysis/passes.

so I pretty much got it from `bazel query` on that package. It contains a bit more analyzers than the ones used in `go vet` (which from the help document I noted in the PR summary).

Reply:
Okay, could you add a comment to this list explaining the difference?